src/v0/destinations/active_campaign/util.js

```diff
@@ -1,31 +1,25 @@
 const { NetworkError } = require('@rudderstack/integrations-lib');
 const {
-  nodeSysErrorToStatus,
   getDynamicErrorType,
+  processAxiosResponse,
 } = require('../../../adapters/utils/networkUtils');
 const tags = require('../../util/tags');
 
 const errorHandler = (err, message) => {
-  if (err.response) {
-    throw new NetworkError(
-      `${message} (${err.response?.statusText},${JSON.stringify(err.response?.data)})`,
-      err.status,
-      {
-        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(err.status),
-      },
-      err,
-    );
-  } else {
-    const httpError = nodeSysErrorToStatus(err.code);
-    throw new NetworkError(
-      `${message} ${httpError.message}`,
-      httpError.status,
-      {
-        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(httpError.status),
-      },
-      err,
-    );
+  const {response, status} = processAxiosResponse(err);
+  const stringifiedResponse = JSON.stringify(response);
+  let msg = `${message} ${stringifiedResponse}`;
+  if (response) {
+    msg = `${message} (${err.response?.statusText},${stringifiedResponse})`;
```

Comment:
We're adding only `statusText` additionally to the error message if `response` (which is originally the `data` field from the raw Axios response) is defined.

Is this accurate or should we change the condition itself?

Reply:
> Is this accurate

I think `Yes`

Reply:
Why don't we directly check if `statusText` exists?

Reply:
We have basically done a fix on top of existing source-code, I wouldn't prefer changing the conditional logic. We can consider this change as part of a refactoring task

Reply:
Since you've agreed that it's a thing to clean up, please resolve this comment once you've captured this as a scope in a different task.